bgpd/bgp_routemap.c

```diff
@@ -421,29 +407,12 @@ route_match_script(void *rule, const struct prefix *prefix, void *object)
 	case LUA_RM_MATCH_AND_CHANGE:
 		status = RMAP_MATCH;
 		zlog_debug("Updating attribute based on script's values");
-
-		uint32_t locpref = 0;
-		struct attr *newattr = frrscript_get_result(fs, &results[1]);
-
-		path->attr->med = newattr->med;
-
-		if (path->attr->flag & ATTR_FLAG_BIT(BGP_ATTR_LOCAL_PREF))
-			locpref = path->attr->local_pref;
-		if (locpref != newattr->local_pref) {
-			SET_FLAG(path->attr->flag,
-				 ATTR_FLAG_BIT(BGP_ATTR_LOCAL_PREF));
-			path->attr->local_pref = newattr->local_pref;
-		}
-
-		aspath_free(newattr->aspath);
-		XFREE(MTYPE_TMP, newattr);
```

Comment:
Is all of this logic supposed to be contained in the lua script now? I don't see it reproduced elsewhere.

Notice that there's extraneous processing bgpd needs to do if these values have changed in order to update them appropriately - this is logic a script author won't and shouldn't need to know about. So we need to retain this in some way unless I missed something.

Reply:
> Is all of this logic supposed to be contained in the lua script now? I don't see it reproduced elsewhere.

Yup, that's the intention: modifications to `attrs['metric']` or `attrs['localpref']` in Lua would have already updated `peer->attr->med` and `peer->attr->local_pref` respectively in C. The last bit of logic missing is setting path->attr->flag if the newattr_localpref is non-zero